test/src/test/java/org/corfudb/integration/Harness.java

```diff
@@ -1,34 +1,81 @@
 package org.corfudb.integration;
 
-import org.corfudb.integration.AbstractIT;
+import static org.corfudb.AbstractCorfuTest.PARAMETERS;
+import static org.corfudb.integration.AbstractIT.PROPERTIES;
+import static org.corfudb.integration.AbstractIT.getCorfuServerLogPath;
+
+import lombok.Getter;
 import org.corfudb.integration.cluster.Harness.Action;
 import org.corfudb.integration.cluster.Harness.Node;
 import org.corfudb.runtime.BootstrapUtil;
+import org.corfudb.runtime.CorfuRuntime;
 import org.corfudb.runtime.view.Layout;
 
 import java.io.IOException;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.UUID;
 
-import static org.corfudb.AbstractCorfuTest.PARAMETERS;
-import static org.corfudb.integration.AbstractIT.getCorfuServerLogPath;
-
 /**
  * Utilities to form and manipulate a corfu cluster.
  * <p>
  * <p>Created by maithem on 7/18/18.
  */
 
+@Getter
 public class Harness {
 
-    String localAddress = "localhost";
+    private int corfuPort;
+    private String corfuHost;
+    private String corfuEndpoint;
+
+    private boolean tlsEnabled;
+    private String runtimePathToKeyStore;
+    private String runtimePathToKeyStorePass;
+    private String runtimePathToTrustStore;
+    private String runtimePathToTrustStorePass;
+    private String serverPathToKeyStore;
+    private String serverPathToKeyStorePassword;
+    private String serverPathToTrustStore;
+    private String serverPathToTrustStorePassword;
+    private int connectionTimeout;
```

Comment:
Looks like all your fields here are immutable, would be great if you make it final

Reply:
You could even use lombook's @Builder to provide nice build api

Reply:
Done